### ISSUE TITLE
Github Actions CI migration

### ISSUE THREAD

**Issue by rzetter** [bfieldtools/bfieldtools#32]
**TL;DR: Travis CI is no longer free for open source projects. Instead you get a free trial good for 1k build minutes, and can email them to beg for more when those run out.**

Starting today Travis will apparently change their pricing, making their 'free' plan into a trial-only affair. For open-source projects, one has to apply for "OSS-only credits" "that will be reviewed and allocated on a case by case basis". In other words, every time the OSS-only credits run out, one has to re-apply, there is no monthly top-up or similar.

Read more here: https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing

The most relevant part for this project:

> We love our OSS teams who choose to build using TravisCI and we fully want to support that community. However, recently we have encountered significant abuse of the intention of this offering. Abusers have been tying up our build queues and causing performance reductions for everyone. In order to bring the rules back to fair playing grounds, we are implementing some changes for our public build repositories.
> - For those of you who have been building on public repositories (on travis-ci.com, with no paid subscription), we will upgrade you to our trial (free) plan with a 10K credit allotment (which allows around 1000 minutes in a Linux environment).
> 
> - You will not need to change your build definitions when you are pointed to the new plan
> - When your credit allotment runs out - we’d love for you to consider which of our plans will meet your needs.
> - We will be offering an allotment of OSS minutes that will be reviewed and allocated on a case by case basis. Should you want to apply for these credits please open a request with Travis CI support stating that you’d like to be considered for the OSS allotment. Please include:
> - Your account name and VCS provider (like travis-ci.com/github/[your account name] )
> - How many credits (build minutes) you’d like to request (should your run out of credits again you can repeat the process to request more)
> - Usage will be tracked under your account information so that you can better understand how many credits/minutes are being used
> 

For the time being, the credit system does not seem to work yet. Both my personal account and the bfieldtools organization has credits available, but builds are not running, I am shown the message below

> "Builds have been temporarily disabled for public repositories due to a negative credit balance. Please go to the Plan page to replenish your credit balance or alter your Consume paid credits for OSS setting."

Now the question is, should we start applying for these OSS-only credits and see how things move forward, or should we move to another CI provider that is friendlier towards open-source, e.g. GitHub Actions?

ping @anttimc @jiivana

**Comment by anttimc** [comment]
Github actions would be an obvious choice, since it does not need to be hosted anywhere. Are there any major limitations?

**Comment by rzetter** [comment]
As far as I can see, there are no major limitations. It's only available for public repositories for free (not a problem), and limited to 2000 run minutes/month (not a problem either). I'll start migrating us over to Github Actions. Now Travis seems to work as well, but we are of course slowly using up our finite trial credits. During the migration, I'll keep both working in parallel (except for pushing new versions to PyPI).

**Comment by jiivana** [comment]
Seems good to me!

**Comment by rzetter** [comment]
Finally had time to get Github Actions fully working (there were some issues with running graphical tests). I've deactivated Travis CI, seems like we're good for now!
